### DISCUSSION TITLE
Overreach of limit attribute - Hosts API

### DISCUSSION THREAD

**Discussion by brucelourenco** [CrowdStrike/falconpy#702] [Ideas]
Greatings!

Do anyone can help me to get information beyond the limit attribute of the API? (5000 for example for Hosts API)

I thought using a loop and store de values in a list, dict or some variable, but the difficulty is how make my requests get the remaining records and not duplicated after the first index of the loop.

Some ideas?
Thank you

**Comment by jshcodes** [comment]
Hi @brucelourenco -

Fantastic timing! I just posted a sample that uses [QueryDevicesByFilterScroll](https://www.falconpy.io/Service-Collections/Hosts.html#querydevicesbyfilterscroll) to produce a list of all sensor versions within a tenant. This [sample](https://github.com/CrowdStrike/falconpy/blob/main/samples/hosts/sensor_versions_by_hostname_scrolling.py) should demonstrate what you're trying to accomplish.

Let us know if you have any questions!  😄

**Reply by brucelourenco** [reply]
Very nice @jshcodes !
I'll try as soon as I can and I'll share here the results.
Thanks a lot!
